src/awkward/_connect/numpy.py

```diff
@@ -420,41 +420,14 @@ def action(inputs, **ignore):
 
         return None
 
-    if sum(int(isinstance(x, ak.contents.Content)) for x in inputs) == 1:
-        where = None
-        for i, x in enumerate(inputs):
-            if isinstance(x, ak.contents.Content):
-                where = i
-                break
-        assert where is not None
-
-        nextinputs = list(inputs)
-
-        def unary_action(layout, **ignore):
-            nextinputs[where] = layout
-            result = action(tuple(nextinputs), **ignore)
-            if result is None:
-                return None
-            else:
-                assert isinstance(result, tuple) and len(result) == 1
-                return result[0]
-
-        out = ak._do.recursively_apply(
-            inputs[where],
-            unary_action,
-            behavior,
-            function_name=ufunc.__name__,
-            allow_records=False,
-        )
```

Comment:
> @jpivarski in this PR so-far, I removed the unary pathway for ufuncs. I can't see why this was introduced in [0e0202a](https://github.com/scikit-hep/awkward/commit/0e0202a5cf54b5dbc2606861d058665b8d90fa07)

That commit is a PR merge; for finer granularity, the commit within that PR is

https://github.com/scikit-hep/awkward/pull/1183/commits/5b0a273e807a8b25ba8b02d2651af28f267823d7#diff-e7b6632c2ca4228bf21505ced52be60d4fda8813e00cbaf65cd1c5d2e228d721R242-R263

It was accompanied by a large batch of single-line assertions, some of which have only one `Content` in the ufunc (such as `+` with one `Content` and one Python number).

Broadcasting a single layout should be equivalent to `recursively_apply` on that one layout. The fact that numerous tests were probably going through that and they passed either way is good evidence that the "should be" holds. (I don't see anything special about it, that it has to be there, and the context of the commit isn't reminding me of any gotchas.)